File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int n = nums.length;
        int arr[] = new int[2];
        for(int i=0; i<n; i++){
            int temp = target-nums[i];
            if(map.containsKey(temp)){
                arr[0]=i;
                arr[1]=map.get(temp);
                return arr;
            }
            map.put(nums[i], i);
        }
        return new int[]{-1,-1};
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """

        freq = {}
        for i, num in  enumerate(nums):
            complement = target-num
            if complement in freq:
                return [freq[complement], i]
            freq[num] = i

        return [-1,-1]

        